main/src/ca/uwaterloo/flix/api/lsp/provider/HoverProvider.scala

```diff
@@ -78,10 +79,11 @@ object HoverProvider {
 
   private def hoverType(tpe: Type, loc: SourceLocation, current: Boolean)(implicit index: Index, root: Root, flix: Flix): JObject = {
     val minTpe = minimizeType(tpe)
+    val boundsRep = SetFormula.restrictableEnumBounds(minTpe).getOrElse("")
     val markup =
       s"""${mkCurrentMsg(current)}
          |```flix
-         |${FormatType.formatType(minTpe)}
+         |$boundsRep${FormatType.formatType(minTpe)}
```

Comment:
I dont like that you call into some function and rely on it to return the same string for non-restrictable enums. 

Why not just insert an extra string after the call to FormatType?

Reply:
It returns `None` for non-restrictable enums, so `$boundsRep` is just `""`. The String of `FormatType` is untouched. The awkwardness is just to avoid extra newlines on non-restrictable enums